spec/src/main/asciidoc/ch03-entity-operations.adoc

```diff
@@ -6122,6 +6122,30 @@ _?1_ ”).
 
 The use of positional parameters is not supported for criteria queries.
 
+==== Arguments to query parameters
+
+Arguments are assigned to query parameters by calling `Query.setParameter()`.
+The first parameter of `setParameter()` identifies the named or positional
+parameter of the query.
+
+An argument may be assigned to a single-valued parameter of a JPQL or
+native SQL query by passing the argument to the second parameter of
+`setParameter()`.
+
+[source,java]
+----
+query.setParameter("name", name)
+----
+
+A list of arguments may be assigned to a collection-valued parameter of
+a JPQL query by passing the arguments in an instance of `java.util.List`
+to the second parameter of `setParameter()`. The list may be empty.
+
+[source,java]
+----
+query.setParameter("names", List.of(name1, name2, name3))
+----
```

Comment:
What about _null_? That can be both - a single value as well as collection value.

Reply:
Does:

> by passing the arguments in a **non-null** instance of `java.util.List`

fix the problem?

Reply:
yes, that should guarantee that _null_ is always treated as a _single_value_ and the lowest boundary for _collection_value_ is _empty_ (never _null_)

Reply:
Done.